documentation/api/query/v4/upgrading-from-v3.markdown

```diff
@@ -0,0 +1,128 @@
+---
+title: "PuppetDB 3.0 » API » v4 » Upgrading to PuppetDB 3.0"
+layout: default
+canonical: "/puppetdb/latest/api/query/v4/upgrading-from-v3.html"
+---
+
+This document describes changes that users will need to be aware of
+to make their code compliant with the breaking changes in PuppetDB 3.0. Most of these
+changes are around the v4 query API, which has been marked 'experimental' since
+2.0.0 but will be the only API available in 3.0. This is not a changelog
+for PuppetDB 3.0, since some of the changes described are present in 2.x.x.
+Each change is marked with the corresponding release version.
+
+### CHANGES THAT MAY BREAK YOUR CODE
+
+* (3.0) All previously dash-separated field names (e.g receive-time), subquery
+  operators (e.g select-facts), and query parameters (e.g order-by) are now
+  underscore-separated. This change does not apply to dash-separated endpoint
+  names such as `aggregate-event-counts`.
+
+* (3.0) We have renamed the "name" key of the catalogs endpoint to "certname", for
```

Comment:
We should really index the following off the endpoint. That is, add a 4 layer title for say /v4/catalogs, and batch all the related changes under that for fast lookup.